src/query.js

```diff
@@ -476,40 +476,38 @@ export default class Query {
    * @returns {Query} The query.
    */
   matches(field, regExp, options = {}) {
+    const flags = [];
+
     if (!isRegExp(regExp)) {
       regExp = new RegExp(regExp);
     }
 
-    if ((regExp.ignoreCase || options.ignoreCase) && options.ignoreCase !== false) {
-      throw new QueryError('ignoreCase flag is not supported.');
-    }
-
     if (regExp.source.indexOf('^') !== 0) {
-      throw new QueryError('regExp must have `^` at the beginning of the expression ' +
-        'to make it an anchored expression.');
+      throw new QueryError('regExp must have \'^\' at the beginning of the expression'
```

Comment:
It'd be nice for this error to say something about the fact that currently, only anchored expressions are supported on Kinvey.